[WIP] Rubocop changes

janxious opened the pull request:
first pass of rubocop same-making, config copied from acts_as_archival

## TODO

- [ ] deal with remaining rubocop todos (maybe)
- [ ] testing

janxious committed:
first pass of rubocop same-making, config copied from acts_as_archival

mediocretes commented on lib/instrumental/reporters/memcache.rb:
```diff
@@ -1,6 +1,7 @@
   class Memcache < Reporter
-    METHODS = [:get, :delete, :cas, :prepend, :append, :replace, :decrement, :increment, :add, :set]
+
+    METHODS = %i[get delete cas prepend append replace decrement increment add set].freeze
```
I kinda... hate %i and %w now.  Thanks, [consulting client].

mediocretes commented on .rubocop.yml:
```diff
@@ -0,0 +1,81 @@
+# TODO: replace the [] below with default once a new version of
+#       rubocop comes out, since there is a PR for it:
+#       https://github.com/bbatsov/rubocop/pull/4038
+Style/PercentLiteralDelimiters:
```
I don't like these rules.

janxious replied:
Feel free to update with your desires or tell me them.

mediocretes replied:
I've expressed my opinion via an edit to .rubocop.yml.

janxious replied:
👍

Note that I just copied these over from acts_as_archival. Feel free to change the hash syntax too. My opinion was we should have a style instead of what that style should be.

mediocretes committed:
Remove %i %w enforcement

These shortcuts (%w[foo bar] instead of ["foo", "bar"]) are not really something we use a lot.  I also notice that hash syntax enforcement is right next to his change, but I'm trying to control myself.

janxious committed:
Merge remote-tracking branch 'origin/master' into rubocop

* origin/master:
  use the gauge proxy for Net::HTTP reporting
  Update request_timing.rb
  basic Net:HTTP tracking
  add some documentation about expected redis metrics
  fix some issues with the abailability of verious pieces of the response for measuring responsee size
  begin documenting the metrics we record by default
  rudimentary rails content length getting for non-asset pieces
  add a logger= proxy for the agent
  prefixed is when there is a prefix, not when there is not a prefix
  route all calls to gauge/increment from reporters through proxy in InstrumentalReporters base so prefixing happens
  add a prefix option for reporter
  rails 5 likes modules instead of strings for middleware
  deal with assets paths in rails dev mode, since it's a known path we were previously reporting as unknown, but it doesn't go through a controller so it needs special handling
  honeybadger does not report everything through the 'Honeybadger' singleton, particularly the rack middleware directly instantiates its own agent and uses it.
  instrumental reporting in ensure block in honeybadger reporter
  remove extra middleware metrics becuaase they have low value. Fixes #4
  modern honeybadger has gotten rid of notify_or_ignore

janxious committed:
fix some stuff from rubocop

janxious committed:
update rubocop todos

janxious committed:
Merge remote-tracking branch 'origin/rubocop' into rubocop

* origin/rubocop:
  Remove %i %w enforcement

janxious committed:
re-run bundle exec rubocop --auto-gen-config --auto-correct
